src/shared-components/alert/index.tsx

```diff
@@ -114,77 +102,65 @@ export class Alert extends React.Component<AlertProps, AlertState> {
-
-    window.setTimeout(() => {
-      this.setState({ exited: true });
+    // ComponentWillUnmount cleanup
+    return () => {
+      if (timer) {
+        window.clearTimeout(timer);
+      }
+    };
+  }, []);
 
-      onExit({ ...rest });
-    }, ANIMATION_DELAY);
-  };
+  if (exited) {
+    return null;
+  }
 
-  render() {
-    const {
-      avatarSrc,
-      content,
-      ctaContent,
-      truncateText,
-      type,
-      // need to destructure onExit to avoid passing as AlertContainer attribute with ...rest
-      // eslint-disable-next-line @typescript-eslint/no-unused-vars
-      onExit,
-      ...rest
-    } = this.props;
-    const { exiting, exited } = this.state;
-    const Icon = alertIconMapping[type];
-
-    if (exited) {
-      return null;
-    }
+  const Icon = alertIconMapping[type];
+
+  return (
+    <AlertContainer
+      alertType={type}
+      exiting={exiting}
+      onClick={alertExitHandler}
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      {...rest}
+    >
+      <MainContainer>
+        <IconContainer hasAvatar={!!avatarSrc}>
+          {avatarSrc ? (
+            <Avatar size="small" src={avatarSrc} alt="avatar" />
+          ) : (
+            <Icon fill={COLORS.white} />
+          )}
+        </IconContainer>
+        <ContentContainer truncateText={truncateText} ref={contentText}>
+          {content}
+        </ContentContainer>
+      </MainContainer>
+      {ctaContent && (
+        <CtaContent>
+          <div>{ctaContent}</div>
+          <ChevronIcon fill={COLORS.white} width={14} height={14} />
+        </CtaContent>
+      )}
+    </AlertContainer>
+  );
+};
 
-    return (
-      <AlertContainer
-        alertType={type}
-        exiting={exiting}
-        onClick={this.alertExitHandler}
-        // eslint-disable-next-line react/jsx-props-no-spreading
-        {...rest}
-      >
-        <MainContainer>
-          <IconContainer hasAvatar={!!avatarSrc}>
-            {avatarSrc ? (
-              <Avatar size="small" src={avatarSrc} alt="avatar" />
-            ) : (
-              <Icon fill={COLORS.white} />
-            )}
-          </IconContainer>
-          <ContentContainer truncateText={truncateText} ref={this.contentText}>
-            {content}
-          </ContentContainer>
-        </MainContainer>
-        {ctaContent && (
-          <CtaContent>
-            <div>{ctaContent}</div>
-            <ChevronIcon fill={COLORS.white} width={14} height={14} />
-          </CtaContent>
-        )}
-      </AlertContainer>
-    );
-  }
-}
+Alert.Container = ({ children }: { children: React.ReactNode }) => (
+  <AlertsContainer>{children}</AlertsContainer>
+);
+
+Alert.propTypes = {
+  avatarSrc: PropTypes.string,
+  content: PropTypes.node.isRequired,
+  ctaContent: PropTypes.node,
+  duration: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
+  onExit: PropTypes.func,
+  truncateText: PropTypes.bool,
+  type: PropTypes.oneOf(['success', 'error', 'default']),
```

Comment:
[Nit]: We have `danger` in `AlertType` but not here, we should add it here as well.

Reply:
Yeah there was a reason for that, danger is not supported anymore but there are still alerts populating from Backend that may have that type (that I couldnt fix) So danger was left there as a failsafe mechanism since the default alert is default type. I guess I agree to add it now since when we rely on TS people will see the danger type anyway. This needs to be solved in a different PR along with the other components like Banner